docs/source/daml-integration-kit/index.rst

```diff
@@ -0,0 +1,375 @@
+  contains interfaces abstracting over the state of
+  a participant node relevant for a DAML Ledger API server. These are the
+  interfaces whose implementation is specific to a particular `X` ledger. These
+  interfaces are optimized for ease of implementation.
+``participant-state-index.jar`` (`source code <https://github.com/digital-asset/daml/tree/master/ledger/participant-state-index>`__)
+  contains code for reading the abstract state
+  of a participant node and indexing it to satisfy the read access
+  patterns required for serving the Ledger API. The library provides both
+  an interface enumerating all read access methods and an in-memory
+  reference implementation of that interface.
+  We expect to provide a persistent, SQL-backed index in the future
+  (`tracking GitHub issue <https://github.com/digital-asset/daml/issues/581>`__).
+``api-server-damlonx.jar`` (`source code <https://github.com/digital-asset/daml/blob/master/ledger/api-server-damlonx/src/main/scala/com/daml/ledger/api/server/damlonx/Server.scala>`__)
+  contains code that implements a DAML Ledger API
+  server given implementations of the interfaces in ``participant-state.jar``.
+``daml-engine.jar`` (`source code <https://github.com/digital-asset/daml/blob/master/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala>`__)
+  contains code for serializing and deserializing DAML
+  transactions and for validating them. An `<X>-daml-validator` is typically
+  implemented by wrapping this code in the `X`-ledger's SDK for building
+  transaction validators. ``daml-engine.jar`` also contains
+  code for interpreting commands sent over the Ledger API. It is used
+  by the `daml-on-<X>-server` to construct the transactions submitted
+  to its participant node.
+
+The following diagram shows how the classes and interfaces provided by these
+libraries are typically combined to instantiate a DAML Ledger API server
+backed by an `X` ledger.
+
+.. image:: images/server-classes-and-interfaces.svg
+
+.. original: https://www.lucidchart.com/invitations/accept/04239d8e-70ec-4734-b943-9780731fa704
+
+The diagram uses boxes labeled with fully qualified class names to denote class instances.
+It uses solid arrows labeled with fully qualified interface names to denote that an instance
+depends on another instance providing that interface. It uses dashed arrows to
+denote that a class instance provides or depends on particular services. It
+uses boxes embedded in other boxes to denote that the outer class instance
+creates the contained instances.
+
+We explain the elements of the diagram in turn. For brevity, we drop prefixes
+of their qualified names where unambiguous.
+
+:doc:`/app-dev/ledger-api-introduction/index`
+  is the collection of gRPC
+  services that you would like your `daml-on-<X>-server` to provide.
+``<X> services``
+  are the services provided by our underlying ledger,
```

Comment:
`our underlying ledger` is confusing. My understanding is that all `X` things are provided by implementors. 
```suggestion
  are the services which underlie your ledger,
```

Reply:
done